likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js

```diff
@@ -0,0 +1,214 @@
+import {
+  useState, useEffect, useMemo,
+} from 'react';
+import axios from 'axios';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { debounce } from 'lodash';
+import Section from '../components/Section';
+import LikecoinInfoTable from '../components/LikecoinInfoTable';
+import SubmitButton from '../components/SubmitButton';
+import LikeButtonPreview from '../components/LikeButtonPreview';
+import SettingNotice from '../components/SettingNotice';
+import LikecoinHeading from '../components/LikecoinHeading';
+import { SITE_LIKER_INFO_STORE_NAME } from '../store/site-likerInfo-store';
+import { USER_LIKER_INFO_STORE_NAME } from '../store/user-likerInfo-store';
+
+function LikecoinButtonPage() {
```

Comment:
After revisiting this, I think it is better to separate this component into a pure component and inject the data from the store through props using a higher-order component

Reply:
Those useEffects are confusing to me

Reply:
By injecting through props, those `useEffects` can be omitted

Reply:
As discussed. Changed in the next comment. :)